src/js/node/events.js

```diff
@@ -92,9 +92,28 @@ const emitWithoutRejectionCapture = function emit(type, ...args) {
   if (events === undefined) return false;
   var handlers = events[type];
   if (handlers === undefined) return false;
-
-  for (var handler of [...handlers]) {
-    handler.apply(this, args);
+  // clone handlers array if necessary since handlers can be added/removed during the loop
+  const maybeClonedHandlers = handlers.length > 1 ? handlers.slice() : handlers;
```

Comment:
```suggestion
  const clonedHandlers = handlers.slice();
```
The user can always modify the handlers. We should always slice.

Reply:
In which case would this be an issue? If we have just one handler attached and this handler is called during the for loop (over the array with length 1), it can either
1) remove itself from the `handlers` array, e.g. if it was attached via .once(). -> No problem since no following handlers need to be called which would be affected by the change in array length.
2) or add additional handlers to the `handlers` array -> also not an issue in my opinion, since the newly added handlers are not called due to the for loop of length 1.
3) or modify itself somehow -> same as for 2) I would assume those modifications should only be considered the next time the emit is triggered.

Am I missing something?

Reply:
Really good point @yschroe! Can you add a code comment above it explaining that for future "us" to know about that.

Reply:
> Really good point @yschroe! Can you add a code comment above it explaining that for future "us" to know about that.

Sure! Is it clear this way?
```
// clone handlers array if necessary since handlers can be added/removed during the loop.
// cloning can be skipped in the case of exactly one attached handler since array length
// changes have no side-effects in a for-loop of length 1
```

Reply:
Yes, thank you!
> // cloning the handlers array is necessary since handlers can be added/removed during the loop.
// cloning can be skipped in the case of exactly one attached handler since array length
// changes have no side-effects in a for-loop of length <= 1